lib/rack/tracker.rb

```diff
@@ -55,7 +55,11 @@ def html?; @headers['Content-Type'] =~ /html/; end
 
     def inject(env, response)
       @handlers.each(env) do |handler|
-        response.gsub!(%r{</#{handler.position}>}, handler.render + "</#{handler.position}>")
+        if handler.container_position == :closing
+          response.gsub!(%r{</#{handler.container_tag}>}, handler.render + "</#{handler.container_tag}>")
```

Comment:
Then what do you think about the grouping idea?

And yeah, good catch, that affects the code 2 lines below! I'll change it and add a spec for that.